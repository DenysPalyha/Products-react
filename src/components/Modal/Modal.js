import React from "react";
import PropTypes from "prop-types";
import styles from "./Modal.module.scss";

const Modal = ({ header, text, closeButton, actions, closeModale }) => {
  return (
    <div className={styles.modal} data-testid="modal-root">
      <div className={styles.bgColorModal} onClick={closeModale}></div>
      <div className={styles.container}>
        <div className={styles.headerModal}>
          <h2 className={styles.headerTitle}>{header}</h2>
          {closeButton && (
            <button className={styles.btnClose} onClick={closeModale}></button>
          )}
        </div>
        <div className={styles.modalBody}>{text}</div>
        <div className={styles.footerBtn}>{actions}</div>
      </div>
    </div>
  );
};

Modal.propTypes = {
  header: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  actions: PropTypes.node.isRequired,
  closeModale: PropTypes.func.isRequired,
  closeButton: PropTypes.bool,
};

Modal.defaultProps = {
  closeButton: true,
};

export default Modal;
