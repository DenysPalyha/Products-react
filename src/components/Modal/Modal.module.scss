.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
}

.bgColorModal {
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  cursor: pointer;
}

.container {
  width: 500px;
  position: relative;
  border-radius: 5px;
  z-index: 2;
  background: #e74c3c;
}

.headerModal {
  padding: 20px 30px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: rgba(0, 0, 0, 0.1);
}

.headerTitle {
  margin: 0;
  font-size: 20px;
  line-height: 1;
  color: #fff;
  font-family: sans-serif;
}

.btnClose {
  width: 30px;
  height: 30px;
  position: relative;
  margin: 0;
  padding: 0;
  outline: none;
  border: none;
  background: transparent;
  cursor: pointer;

  &::after,
  &::before {
    content: "";
    position: absolute;
    height: 1px;
    width: 100%;
    top: 50%;
    left: 0;
    background: rgb(255, 255, 255);
    border-radius: 5px;
  }

  &::after {
    transform: (rotate(45deg));
  }
  &::before {
    transform: (rotate(-45deg));
  }
}

.modalBody {
  padding: 30px 40px;
  font-size: 18px;
  line-height: 2;
  color: #fff;
  text-align: center;
}

.footerBtn {
  text-align: center;
  margin-bottom: 20px;
}
