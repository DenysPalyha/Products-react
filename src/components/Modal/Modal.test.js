import { render } from "@testing-library/react";

import Modal from "./Modal";

describe("Modal snapshot component", () => {
  test("should Modal must match snapshot", () => {
    const { asFragment } = render(
      <Modal
        header="modal test"
        text="text test"
        closeButton={true}
        actions={<button>click test</button>}
        closeModale={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
