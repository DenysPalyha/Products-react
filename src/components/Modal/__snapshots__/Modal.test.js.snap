// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot component should Modal must match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
    data-testid="modal-root"
  >
    <div
      class="bgColorModal"
    />
    <div
      class="container"
    >
      <div
        class="headerModal"
      >
        <h2
          class="headerTitle"
        >
          modal test
        </h2>
        <button
          class="btnClose"
        />
      </div>
      <div
        class="modalBody"
      >
        text test
      </div>
      <div
        class="footerBtn"
      >
        <button>
          click test
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
